src/fides/api/ctl/database/samples.py

```diff
@@ -0,0 +1,151 @@
+            # that for the sample_resources/*, however, to stay consistent with
+            # the behaviour of `fides push`
+            yaml_dict = yaml.safe_load(file)
+
+            # Parse the contents of the YAML into the respective Pydantic models
+            if yaml_dict.get("dataset", []):
+                datasets = [Dataset.parse_obj(e) for e in yaml_dict.get("dataset")]
+                sample_resources_dict["dataset"].extend(datasets)
+            elif yaml_dict.get("organization", []):
+                organizations = [
+                    Organization.parse_obj(e) for e in yaml_dict.get("organization")
+                ]
+                sample_resources_dict["organization"].extend(organizations)
+            elif yaml_dict.get("policy", []):
+                policies = [Policy.parse_obj(e) for e in yaml_dict.get("policy")]
+                sample_resources_dict["policy"].extend(policies)
+            elif yaml_dict.get("system", []):
+                systems = [System.parse_obj(e) for e in yaml_dict.get("system")]
+                sample_resources_dict["system"].extend(systems)
+            else:
+                if strict:
+                    unknown_resource_keys = list(yaml_dict.keys())
+                    raise NotImplementedError(
+                        f"Unexpected sample resource type, keys={unknown_resource_keys}"
+                    )
+                # quietly ignore failures when strict=False, so we only fail during tests
+    return sample_resources_dict
+
+
+class SampleConnection(CreateConnectionConfigurationWithSecrets):
+    """
+    Schema for creating a sample connection to initialize the database, usable
+    for both database & SaaS connections. Extends the existing models with an
+    optional 'dataset' field.
+    """
+
+    dataset: Optional[FidesKey] = None
+    """
+    (Optional) Valid fides_key for an existing dataset that should be used
+    for this ConnectionConfig. If a dataset does not exist for this key,
+    creation will fail.
+    """
+
+
+def load_sample_connections_from_project() -> List[SampleConnection]:
+    """
+    Loads all the sample connection YAML files from the sample project by
+    traversing through the sample_connections/ folder, inspecting each file, and
+    parsing them into "ConnectionConfig" models that would be suitable for the
+    `PATCH /api/v1/connection` API.
+
+    While parsing the YAML file, this also performs ENV variable expansion,
+    similar to what Docker/Bash/etc. support. This allows secrets to be provided
+    as ENV var names (e.g. $FIDES_DEPLOY__CONNECTORS__POSTGRES__PASSWORD) and
+    replaced with "real" values at runtime.
+
+    After parsing the models, this function then *excludes* any connection missing
+    any of it's secrets. This prevents gotchas with failing connections at runtime
+    and also makes it easy to programmatically control which connections should be
+    enabled by simply adding/removing the secrets to the config file.
```

Comment:
Both of these notes are pretty key - this helper will use `expandvars()` to insert secrets at runtime, and if any connection has undefined secrets they'll be ignored.

This means that if you run `fides deploy` with `FIDES_DEPLOY__CONNECTORS__MAILCHIMP_API_KEY` unset, the Mailchimp connection isn't created.